### PR TITLE
switch to ctypes

### PR DESCRIPTION
**Pull request by jkanche**

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -1,2 +1,3 @@
-recursive-include src/mattress/lib/ *
-recursive-include extern *
\ No newline at end of file
+recursive-include src/mattress/extern * 
+recursive-include src/mattress/include * 
+recursive-include src/mattress/lib *
```
Comment (LTLA) [inline MANIFEST.in:3]: You don't want to include the `*.cpp` sources, as these are irrelevant once the shared library is generated.

File: src/mattress/cpphelpers.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+# TODO: surely there's a better way than whatever this is.
+import os
+
+__author__ = "ltla, jkanche"
+__copyright__ = "ltla, jkanche"
+__license__ = "MIT"
+
+
+def includes() -> list[str]:
+    """includes provides access libraries for downstream apps.
```
Comment (LTLA) [inline src/mattress/cpphelpers.py:12]: "provides access to C++ headers (including tatami) for downstream packages."

File: src/mattress/cpphelpers.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    Returns:
+        list[str]: list of paths to the source files.
```
Comment (LTLA) [inline src/mattress/cpphelpers.py:15]: headers, not source files.

File: setup.cfg [hunk 4]
```diff
@@ -49,7 +49,6 @@ package_dir =
 # For more information, check out https://semver.org/.
 install_requires =
     importlib-metadata; python_version<"3.8"
-    Cython
     numpy>=1.22.4
     scipy
```
Comment (LTLA) [inline setup.cfg:53]: also, why is scipy here?

File: MANIFEST.in (hunk 1 above)
Comment (jkanche) [reply MANIFEST.in:3]: it fails if i don't include them, atleast on source based builds

File: setup.cfg (hunk 4 above)
Comment (jkanche) [reply setup.cfg:53]: i probably just included it to support sparse

**Pull request merged**